cucumber-core/src/main/java/io/cucumber/core/eventbus/FastUuidGenerator.java

```diff
@@ -0,0 +1,32 @@
+package io.cucumber.core.eventbus;
+
+import io.cucumber.core.exception.CucumberException;
+
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Thread-safe UUID generator for single JVM. This is a sequence generator and
+ * each instance has its own counter. This generator is about 100 times faster
+ * than #DefaultUuidGenerator. If you use Cucumber in multi-JVM setup, you
+ * should use #DefaultUuidGenerator instead.
+ */
+public class FastUuidGenerator implements UuidGenerator {
```

Comment:
While it's hard to imagine that a faster generator could be created, I don't think the implementation speed is the property that should be used to name this. I think something like `IncrementingUuidGenerator` would be better.

That is, assuming we don't change the implementation to be something else.